lib/absinthe/resolution.ex

```diff
@@ -200,6 +200,16 @@ defmodule Absinthe.Resolution do
   end
 
   def call(%{state: :unresolved} = res, resolution_function) do
+    res = %{
+      res
+      | acc:
+          Map.put_new(
+            res.acc,
+            {:telemetry, path(res)},
+            {System.system_time(:millisecond), System.monotonic_time()}
+          )
```

Comment:
Ah ok so the start time is only actually set when resolution functions are run. In this case what you should do is have this middleware to do `res = %{res | middleware ++ [EmitResolvedTelemetry]}` or something (find a better name). Then it'll only run telemetry emission for fields that are guaranteed to have the start time recorded, and then it doesn't need to be baked into resolution.

Reply:
Right, this is what I do in the New Relic Absinthe middleware. I baked it in here because I wanted to make it built-in, ie: w/o needing to define a custom middleware stack in your schema.

Is there a way I can build in this middleware into all resolvers? It'd only do anything for `Resolution.call` based resolvers, so it'd skip `Map.get` etc..

https://github.com/binaryseed/new_relic_absinthe/blob/687e4fe7d152ba95e6af4e8b021309658f452d10/lib/new_relic/absinthe/middleware.ex#L6-L26

Reply:
I found a place to stick an extra middleware on each resolver, so I pushed a commit with that. Not sure if that's the right location but it isolates all the Telemetry stuff into a module.